esp/src/eclwatch/TopologyWidget.js

```diff
@@ -18,53 +18,163 @@ define([
+], function (declare, i18n, nlsHPCC, on, dom, domConstruct, lang,
+                CheckBox, TextBox, ValidationTextBox, registry, ToggleButton, Select, ToolbarSeparator, ContentPane,
+                tree, selector,
+                GridDetailsWidget, PreflightDetailsWidget, RequestInformationWidget, ESPRequest, ESPTopology, TopologyDetailsWidget, DelayLoadWidget, ESPUtil, FilterDropDownWidget, WsMachine) {
     return declare("TopologyWidget", [GridDetailsWidget], {
 
         i18n: nlsHPCC,
         gridTitle: nlsHPCC.title_Topology,
         idProperty: "__hpcc_id",
+        filter: null,
+        filterLoaded: null,
 
         postCreate: function (args) {
             this.inherited(arguments);
             this.detailsWidget = new TopologyDetailsWidget({
                 id: this.id + "Details",
                 region: "right",
                 splitter: true,
-                style: "width: 66%",
+                style: "width: 80%",
                 minSize: 240
             });
             this.detailsWidget.placeAt(this.gridTab, "last");
+            this.filter = new FilterDropDownWidget();
         },
 
         init: function (params) {
+            var context = this;
             if (this.inherited(arguments))
                 return;
-
+            this.detailsWidget.requestInformationWidget.set("disabled", true);
+            this.filter.disable(true);
             this.refreshGrid();
+
+            this.filter.on("apply", function (evt) {
+                var selection = context.grid.getSelected();
+                var filter = context.getFilter();
+
+                for (var i = 0; i < selection.length; ++i) {
+                    if (context.viewModeMachines.checked || context.viewModeServices.checked) {
+                        var MachineInformationClean = "Addresses." + [i];
+                        var MachineInformationCount = "Addresses.itemcount";
+                        if (context.viewModeMachines.checked) {
+                            filter[MachineInformationClean] = selection[i].getNetaddress() + "|:" + selection[i].__hpcc_parentNode.__hpcc_treeItem.Type + ":" + selection[i].__hpcc_treeItem.Name + ":" + 2 + ":" + selection[i].__hpcc_treeItem.Directory + ":" + 0;
+                        } else if (context.viewModeServices.checked) {
+                            filter[MachineInformationClean] = selection[i].__hpcc_children[0].__hpcc_treeItem.Netaddress + "|:" + selection[i].__hpcc_treeItem.Type + ":" + selection[i].__hpcc_treeItem.Name + ":" + 2 + ":" + selection[i].__hpcc_children[0].__hpcc_treeItem.Directory;
+                        }
+                            WsMachine.GetMachineInfo ({
+                                request: {
+                                    [MachineInformationClean]: filter[MachineInformationClean],
+                                    [MachineInformationCount]: 1000,
```

Comment:
I verified with @wangkx this count seems to increase based on selections and is subjective.